.github/workflows/tests.yml

```
@@ -34,45 +65,53 @@ jobs:
       with:
         go-version: ${{ env.GO_VERSION }}
 
+    - name: Set up Go environment
+      run: |
+        echo "GOCACHE=$(go env GOCACHE)" >> $GITHUB_ENV
+        echo "GOMODCACHE=$(go env GOMODCACHE)" >> $GITHUB_ENV
+      shell: bash
+
+    - name: Skip additional wrapper CI tests
+      if: matrix.skip_wrapper_tests == true
+      run: |
+        echo "SKIP_WRAPPER_CI_TESTS=true" >> $GITHUB_ENV
+      shell: bash
+
+    - name: Additional steps for 386 architecture
+      if: matrix.goarch == '386'
+      run: |
+        echo "CGO_ENABLED=1" >> $GITHUB_ENV
+        sudo apt-get update
+        sudo apt-get install gcc-multilib g++-multilib
+      shell: bash
+
     - name: Cache go modules  
       uses: actions/cache@v3
       with:
         path: |
-          ~/go/pkg/mod
-          ~/.cache/go-build
-        key: ${{ runner.os }}-go-${{ hashFiles('**/go.sum') }}
-        restore-keys: ${{ runner.os }}-go-
+          ${{ env.GOMODCACHE }}
+          ${{ env.GOCACHE }}
+        key: ${{ runner.os }}-go-${{ matrix.goarch }}-${{ hashFiles('**/go.sum') }}
+        restore-keys: ${{ runner.os }}-go-${{ matrix.goarch }}-
 
     - name: Test
-      run: go test -race -coverprofile coverage.txt -covermode atomic ./...
+      run: |
+        unset PGSERVICEFILE
+        if [ "${{ matrix.goarch }}" = "386" ]; then
+          go test -coverprofile coverage.txt -covermode atomic ./...
+        else
+          go test -race -coverprofile coverage.txt -covermode atomic ./...
+        fi
+      shell: bash
       env:
-        PSQL_USER: postgres
-        PSQL_PASS: postgres
-        PSQL_HOST: localhost
-        PSQL_DBNAME: gct_dev_ci
-        PSQL_SKIPSQLCMD: true
-        PSQL_TESTDBNAME: gct_dev_ci
-        PSQL_SSLMODE: disable
+        GOARCH: ${{ matrix.goarch }}
 
     - name: Upload coverage to Codecov
       uses: codecov/codecov-action@v3
 
-  backend-32bit:
-    name: GoCryptoTrader back-end 32-bit with PSQL
+  backend-arm64:
```

For the sake of consistency, 
```suggestion
  back-end (ubuntu-latest, arm64, false, true)
```

---

You were very close, just had to be set on the name var below so I've pushed a commit to do that 🚀